File: 2025/Day1/part1.cpp
#include <iostream>
#include <fstream>
#include <string>

int rotate(int currentVal, int dist, char direction){
    if(direction == 'L'){
        return ((currentVal - dist) % 100 + 100) % 100;
    }
    else if(direction == 'R'){
        return ((currentVal + dist) % 100 + 100) % 100;
    }
    else{
        throw std::invalid_argument("Invalid direction.");
    }
}

int main() {
    int currentVal = 50;
    int hits = 0;

    std::cout << "Reading input file." << std::endl;
    std::ifstream file("inputValues.txt");
    if (!file.is_open()) {
        std::cerr << "Could not open file\n";
        return 1;
    }

    std::string line;
    while (std::getline(file, line)) {
        //std::cout << line << '\n';
        char dir = line[0];
        //std::cout << "Direction: " << dir << " of type: " << typeid(dir).name() << std::endl;
        int dist = std::stoi(line.substr(1)); 
        if(dir != 'L' && dir != 'R'){
            throw std::invalid_argument("Invalid direction from input file.");
        }
        currentVal = rotate(currentVal, dist, dir);
        if(currentVal == 0){
            hits++;
        }
        //std::cout << "Current Value: " << currentVal << std::endl;
    }
    std::cout << "Amount of times 0 was hit: " << hits << std::endl;

    return 0;
}

File: 2025/Day1/part1.py
def rotate(currentVal: int, dist: int, direction: str):
    if(direction == "L"):
        return (currentVal - dist) % 100
    elif(direction == "R"):
        return (currentVal + dist) % 100
    else:
        raise ValueError("Invalid direction.")

def main():
    with open("inputValues.txt", "r", encoding="utf-8") as f:
        text = f.read()

    currentVal = 50
    hits = 0

    for line in text.splitlines():
        if(line[0] not in ("L", "R")):
            raise ValueError("Direction of turn is not specified")
        else:
            turnDistance = int(line[1:])
        currentVal = rotate(currentVal, turnDistance, line[0])
        if(currentVal == 0):
            hits += 1
    print(f"Amount of times 0 was hit: {hits}")

if __name__ == "__main__":
    main()